nmoscommon/aggregator.py

```diff
@@ -179,37 +182,42 @@ def _process_queue(self):
                         self._mdns_updater.P2P_disable()
         self.logger.writeDebug("Stopping HTTP queue processing thread")
 
-    # Queue a request to be processed. Handles all requests except initial Node POST which is done in _process_reregister
     def _queue_request(self, method, namespace, res_type, key):
+        """Queue a request to be processed. Handles all requests except initial Node POST which is done in _process_reregister"""
         self._reg_queue.put({"method": method, "namespace": namespace, "res_type": res_type, "key": key})
 
-    # Register 'resource' type data including the Node
-    # NB: Node registration is managed by heartbeat thread so may take up to 5 seconds!
     def register(self, res_type, key, **kwargs):
+        """Register 'resource' type data including the Node
+        NB: Node registration is managed by heartbeat thread so may take up to 5 seconds"""
         self.register_into("resource", res_type, key, **kwargs)
 
-    # Unregister 'resource' type data including the Node
     def unregister(self, res_type, key):
+        """Unregister 'resource' type data including the Node"""
         self.unregister_from("resource", res_type, key)
 
-    # General register method for 'resource' types
     def register_into(self, namespace, res_type, key, **kwargs):
+        """General register method for 'resource' types"""
         data = kwargs
         send_obj = {"type": res_type, "data": data}
         if 'id' not in send_obj["data"]:
             self.logger.writeWarning("No 'id' present in data, using key='{}': {}".format(key, data))
             send_obj["data"]["id"] = key
-
         if namespace == "resource" and res_type == "node":
             # Handle special Node type
             self._registered["node"] = send_obj
+
+            if self.auth_reg is None:
```

Comment:
It might need both? I found there were several calls to this function when starting the nodefacade, which resulted in multiple registrations (which to me points in the direction that it might in the wrong place?)